test/test_torch.py

```diff
@@ -6106,6 +6106,116 @@ def test_tensor_shape_empty(self):
             self.assertEqual([(0, 1, 3, 0)], [z.shape for z in torch.split(x, 1, dim=0)])
             self.assertEqual([(0, 1, 3, 0)], [z.shape for z in torch.split(x, 0, dim=0)])
 
+    # functions that operate over a dimension but don't reduce.
+    @skipIfNoZeroSize
+    def test_dim_function_empty(self):
+        # FIXME: enable CUDA tests.
+        devices = ['cpu']  # if not torch.cuda.is_available() else ['cpu', 'cuda']
+        for device in devices:
+            shape = (0, 1, 2, 0)
+            x = torch.randn(shape, device=device)
+
+            # size stride
+            self.assertEqual(0, x.size(3))
+            self.assertEqual(2, x.size(2))
+            self.assertEqual(2, x.stride(0))
+            self.assertEqual(1, x.stride(2))
```

Comment:
Did we document the "stride for zero-size dims" invariant anywhere? If we're going to test that the stride matches exactly, that implies that the stride we select in these situations is part of the public API.

Reply:
Good question, and I'm okay changing this to just check that you get back some reasonable value if we want to do that.  But code like flip_cuda (see this PR) already assumes there is some canonical "stride for contiguous tensors" (and yes, numel() == 0 tensors are considered contiguous).  So this isn't really a departure from that.  Also, we do the same thing as NumPy, and NumPy doesn't document their numel() == 0 strides as far as I can tell.